bot/zephyr.py

```diff
@@ -111,7 +112,7 @@ def build_and_flash(zephyr_wd, board, board_id, conf_file=None):
     cmd_flash = ['west', 'flash', '--snr', board_id]
     check_call(cmd_flash, env=env, cwd=tester_dir)
 
-    return get_tty_path("J-Link")
+    return get_tty_path(board, board_id)
```

Comment:
You could just use Serial number as a string here instead of "J-Link" and it should find the correct device.

Reply:
I think board_id is already passing the serial number parameter here

Reply:
What I mean here is: if board_id is set then pass it to `get_tty_path()` instead of `"J-Link"`. If not, then keep old behavior.